include/Graph/Graph.hpp

```diff
@@ -0,0 +1,74 @@
+#pragma once
+#include <iostream>
+#include <queue>
+#include <stdexcept>
+#include <string>
+#include <vector>
+
+class Layer {
+ private:
+  int id;
+  std::string name;
+  int type;
+  std::string version;
+  int numInputs;
+  int numNeurons;
+  std::vector<int> primer;
+
+ public:
+  Layer() : id(0), type(1) {}
+  Layer(int id1, int type1) : id(id1), type(type1) {}
+  int checkID() { return id; }
+  void In(std::vector<int> a) { primer = a; }
```

Comment:
It could be taken by reference and then copied to the layer

Reply:
id?

Reply:
const std::vector<int>& a

Reply:
done